File: problems-extend-adapt/1601-1700/1646_grown-array-maximum/solution.go
func grownArrayMaximum(n int) int {
	if n == 0 {
		return 0
	}
	nums := make([]int, n+1)
	nums[1] = 1
	best := 1
	for i := 2; i <= n; i++ {
		if i%2 == 0 {
			nums[i] = nums[i/2]
		} else {
			nums[i] = nums[i/2] + nums[i/2+1]
		}
		if nums[i] > best {
			best = nums[i]
		}
	}
	return best
}

File: problems-extend-adapt/1601-1700/1646_grown-array-maximum/solution.js
/**
 * @param {number} n
 * @return {number}
 */
var grownArrayMaximum = function (n) {
    if (n === 0) {
        return 0;
    }
    const nums = new Array(n + 1).fill(0);
    nums[1] = 1;
    let best = 1;
    for (let i = 2; i <= n; i++) {
        if (i % 2 === 0) {
            nums[i] = nums[i / 2];
        } else {
            nums[i] = nums[(i - 1) / 2] + nums[(i - 1) / 2 + 1];
        }
        best = Math.max(best, nums[i]);
    }
    return best;
};
